frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.ts

```diff
@@ -0,0 +1,334 @@
+
+    this.date = this.changeset.value('date');
+  }
+
+  ngAfterViewInit():void {
+    this
+      .dateModalRelations
+      .getMinimalDateFromPreceeding()
+      .subscribe((date) => {
+        this.initializeDatepicker(date);
+        this.onDataChange();
+      });
+
+    this
+      .dateChangedManually$
+      .pipe(
+        // Avoid that the manual changes are moved to the datepicker too early.
+        // The debounce is chosen quite large on purpose to catch the following case:
+        //   1. date is for example 2022-07-15. The user wants to set the day value  to the 19th.
+        //   2. So he/she starts entering the finish date 2022-07-1 .
+        //   3. This is already a valid date. Since it is before the date,the date would be changed automatically to the first without the debounce.
+        //   4. The debounce gives the user enough time to type the last number "9" before the changes are converted to the datepicker and the start date would be affected.
+        //
+        // Debounce delay is 0 for initial display, and then set to 800
+        debounce(() => timer(this.debounceDelay)),
+      )
+      .subscribe(() => {
+        // set debounce delay to its real value
+        this.debounceDelay = 800;
+
+        // Always update the whole form to ensure that no values are lost/inconsistent
+        this.updateDate(this.date);
+      });
+  }
+
+  changeSchedulingMode():void {
+    this.initializeDatepicker();
+    this.cdRef.detectChanges();
+  }
+
+  changeNonWorkingDays():void {
+    this.initializeDatepicker();
+
+    // If we're single date, update the date
+    if (!this.includeNonWorkingDays && this.date) {
+      // Resent the current start and end dates so duration can be calculated again.
+      this.dateUpdates$.next(this.date);
+    }
+
+    this.cdRef.detectChanges();
+  }
+
+  save($event:Event):void {
+    $event.preventDefault();
+    // Apply the changed scheduling mode if any
+    this.changeset.setValue('scheduleManually', this.scheduleManually);
+
+    // Apply the dates if they could be changed
+    if (this.dateModalScheduling.isSchedulable) {
+      this.changeset.setValue('date', mappedDate(this.date));
+    }
+
+    this.closeMe();
+  }
+
+  cancel():void {
+    this.closeMe();
+  }
+
+  updateDate(val:string|null):void {
+    // Expected minimal format YYYY-M-D => 8 characters OR empty
+    if (val !== null && (val.length >= 8 || val.length === 0)) {
+      if (validDate(val) && this.datePickerInstance) {
+        const dateValue = parseDate(val) || undefined;
+        this.enforceManualChangesToDatepicker(dateValue);
+      }
+    }
+  }
+
+  setToday():void {
+    const today = parseDate(new Date()) as Date;
+    this.date = this.timezoneService.formattedISODate(today);
+    this.enforceManualChangesToDatepicker(today);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  reposition(element:JQuery<HTMLElement>, target:JQuery<HTMLElement>):void {
+    element.position({
```

Comment:
Can we do this without jquery?

Reply:
Right now we're using the `jquery-ui position` helper to provide positioning with edge detection. I don't think we have an alternative library right now, but in that sense its code was only moved from the previous op-datepicker-modal. Would you be okay with us adding a code maintenance ticket to look for a replacement? It's being used in these places:

https://github.com/opf/openproject/blob/dev/frontend/src/app/shared/components/op-context-menu/op-context-menu.service.ts#L116

https://github.com/opf/openproject/blob/dev/frontend/src/app/shared/components/datepicker/single-date-modal/single-date.modal.ts#L265

https://github.com/opf/openproject/blob/dev/frontend/src/app/shared/components/datepicker/multi-date-modal/multi-date.modal.ts#L367

https://github.com/opf/openproject/blob/dev/frontend/src/app/shared/components/modals/preview-modal/wp-preview-modal/wp-preview.modal.ts#L83

Reply:
Works for me, I'll create a maintenance ticket. I'm not sure if we need a library for this or if one or two stateful functions that position the DOM element are enough.